examples/00-systemcoupling/fsi.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+""".. _ref_fsi_example:
+
+fsi.py Example
+-------------------------------------------------
+
+This example demonstrates a simple fluid-structure interaction (FSI) simulation
+
+"""
+
+# import required modules
+import shutil
+
+import ansys.fluent.core as pyfluent
+import ansys.mapdl.core as pymapdl
+
+import ansys.systemcoupling.core as pysyc
+from ansys.systemcoupling.core import examples
+
+# Connect to mapdl container
+print("Connecting to MAPDL")
+mapdl = pymapdl.Mapdl()
+
+# ================================
+
+# read in the pre-created Fluent case file and launch Fluent
+
+fluent_cas_file = examples.download_file(
+    "plate.cas.gz", "pysystem-coupling/oscillating_plate/Fluent"
+)
+
+shutil.copy(fluent_cas_file, "plate.cas.gz")
+
+fluent_cas_file = "plate.cas.gz"
+
+custom_config = {"fluent_image": "ghcr.io/ansys/pyfluent:v24.2.0"}
```

Comment:
@ansMHanmer can you remember whether this custom config was necessary for some reason? I will need to be able to remove explicit mention of docker images in Sphinx Gallery examples as the idea is that they should be runnable on the desktop without modification. Hopefully, running in container mode can be configured using environment variables during doc build.

Reply:
It is necessary to instruct pyfluent to connect to a fluent container rather than a desktop installation.  I had another look and don't see any env vars to control this.  We might have to support both, with the default being a desktop installation, but allowing Docker to launch if the `CI` variable is set to true, indicating we're running on GitHub? https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Reply:
I think Fluent itself uses a combination of `PYFLUENT_LAUNCH_CONTAINER=1` and `FLUENT_IMAGE_TAG` in its pipelines in order to start a container seamlessly when `launch()` is used. I will look into it.

Reply:
I did try to get this working on this branch but ran into the issue that I couldn't see the workflow at all to run it. I am not sure why but it might be because that `ci.yml` is currently showing as conflicted. I have started a [new branch ](https://github.com/ansys/pysystem-coupling/pull/340)  from an up to date `main` and have incorporated your changes as well as the things I was trying. The good news is that it seems to work if I set the environment variable I mentioned.

Reply:
Great, thanks @iboyd-ansys !  I've resolved the conflict in my branch for good measure.